retrieval/discovery/kubernetes/pod.go

```diff
@@ -74,7 +74,10 @@ func (p *Pod) Run(ctx context.Context, ch chan<- []*config.TargetGroup) {
 			send(p.buildPod(o.(*apiv1.Pod)))
 		},
 		DeleteFunc: func(o interface{}) {
-			send(&config.TargetGroup{Source: podSource(o.(*apiv1.Pod))})
+			pod, ok := o.(*apiv1.Pod)
```

Comment:
We only need the namespace & name which is available in the (albeit stale) `DeletedFinalStateUnknown.Obj` field?